### PR TITLE
Measure latency of many RedisRaft operations

### PR DESCRIPTION
**Pull request by sjpotter**
adds measurments using the Redis Latency sampling interface

### CODE REVIEW

File: raft.c [hunk 1]
```diff
@@ -149,11 +152,13 @@ static void executeRaftRedisCommandArray(RaftRedisCommandArray *array,
             continue;
         }
 
+        uint64_t begin_rmcall = timeMonotonicNanos();
         enterRedisModuleCall();
         RedisModuleCallReply *reply = RedisModule_Call(
                 ctx, cmd, redis_raft.resp_call_fmt, &c->argv[1], c->argc - 1);
         int ret_errno = errno;
         exitRedisModuleCall();
+        RedisModule_LatencyAddSample("rm_call", (mstime_t) ((timeMonotonicNanos() - begin_rmcall) / 1000000));
```
Comment (tezc) [inline raft.c:161]: Looks like `RedisModule_LatencyAddSample()` has millisecond precision only, so maybe we just use RedisModule_Milliseconds(). It has less precise timer call (gettimeofday(), less precise, better for performance).

File: raft.c [hunk 2]
```diff
@@ -149,11 +152,13 @@ static void executeRaftRedisCommandArray(RaftRedisCommandArray *array,
             continue;
         }
 
+        uint64_t begin_rmcall = timeMonotonicNanos();
```
Comment (tezc) [inline raft.c:155]: I see that inside RedisModule_Call(), it measures time passed. Maybe we avoid doing that here?

File: raft.c [hunk 3]
```diff
@@ -1367,6 +1404,8 @@ void RaftReqFree(RaftReq *req)
 
         RedisModule_UnblockClient(req->client, NULL);
     }
+
+    RedisModule_LatencyAddSample(RaftReqTypeStr[req->type],  (mstime_t) ((timeMonotonicNanos() - req->begin) / 1000000));
```
Comment (tezc) [inline raft.c:1408]: RedisModule_LatencyAddSample() does not seem thread-safe to me. So, I believe we should guarantee that we call it only in raft thread. Seems like we can call RaftReqFree() in redis thread itself? Hope I'm all wrong :)

**Comment by sjpotter** [comment]
not happening

**Pull request closed without merging**